### DISCUSSION TITLE
AirSDK iOS AppStore - Missing Push Notification Entitlement

### DISCUSSION THREAD

**Discussion by william3954** [airsdk/Adobe-Runtime-Support#987] [Q&A] [answered]
When submitting to the app store (iOS), I receive the following email from Apple:

> Dear Developer,
> 
> We identified one or more issues with a recent delivery for your app, "MyAppTitle" 0.3.8 (0.3.8). Your delivery was successful, but you may wish to correct the following issues in your next delivery:
> 
> ITMS-90078: Missing Push Notification Entitlement - Your app appears to register with the Apple Push Notification service, but the app signature's entitlements do not include the 'aps-environment' entitlement. If your app uses the Apple Push Notification service, make sure your App ID is enabled for Push Notification in the Provisioning Portal, and resubmit after signing your app with a Distribution provisioning profile that includes the 'aps-environment' entitlement. Xcode does not automatically copy the aps-environment entitlement from provisioning profiles at build time. This behavior is intentional. To use this entitlement, either enable Push Notifications in the project editor's Capabilities pane, or manually add the entitlement to your entitlements file. For more information, see https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/HandlingRemoteNotifications.html#//apple_ref/doc/uid/TP40008194-CH6-SW1.
> 
> After you’ve corrected the issues, you can upload a new binary to App Store Connect.
> 
> Best regards,
> 
> The App Store Team

I don't believe this would be a big issue to correct by re-provisioning the app, however, it does not use push notifications. Is there a setting within AirSDK that can be changed to not trigger this message?

**Comment by marchbold** [comment, marked as answer]
Are you using any of distriqt's ANE's? I believe this can be caused by adding our centralised comms ANE (Core) which has some push related code in it (as it's used by multiple extensions). It is completely safe to ignore.

**Reply by william3954** [reply]
Thanks. That is the source of the warning.
